readthedocs/builds/forms.py

```diff
@@ -29,9 +32,34 @@ def __init__(self, instance=None, *args, **kwargs):  # noqa
 
 class VersionForm(forms.ModelForm):
 
+    slug = forms.RegexField(
+        '^{pattern}$'.format(pattern=VERSION_SLUG_REGEX),
+        max_length=255,
+        help_text=_("Used in this version's URL"),
+    )
+
     class Meta(object):
         model = Version
-        fields = ['active', 'privacy_level', 'tags']
+        fields = ['slug', 'active', 'privacy_level', 'tags']
+
+    def __init__(self, *args, **kwargs):
+        super(VersionForm, self).__init__(*args, **kwargs)
+        if self.instance and self.instance.slug in (LATEST, STABLE):
+            self.fields['slug'].disabled = True
+            self.fields['slug'].help_text += _(' - read only for special versions')
+
+    def clean_slug(self):
+        slug = self.cleaned_data['slug']
+
+        version = self.instance
+        if version:
+            if Version.objects.filter(project=version.project, slug=slug).exclude(
+                    pk=version.pk).count() > 0:
```

Comment:
Use `.exists()` rather than `.count() > 0`?

Reply:
Thanks. I feel like I do this all the time.